modules/component/src/core/operators/coalesce.ts

```diff
@@ -0,0 +1,216 @@
+import {
+  MonoTypeOperatorFunction,
+  Observable,
+  Operator,
+  SubscribableOrPromise,
+  Subscriber,
+  Subscription,
+  TeardownLogic,
+} from 'rxjs';
+import {
+  InnerSubscriber,
+  OuterSubscriber,
+  subscribeToResult,
+} from 'rxjs/internal-compatibility';
+import { generateFrames } from '../projections';
+import { createPropertiesWeakMap } from '../utils/create_properties-weakmap';
+
+export interface CoalesceConfig {
+  context?: object;
+  leading?: boolean;
+  trailing?: boolean;
+}
+
+interface CoalescingContextProps {
+  isCoalescing: boolean;
+}
+
+const coalescingContextPropertiesMap = createPropertiesWeakMap<
+  object,
+  CoalescingContextProps
+>(ctx => ({
+  isCoalescing: false,
+}));
+
+const defaultCoalesceConfig: Pick<CoalesceConfig, 'leading' | 'trailing'> & {
+  context: undefined;
+} = {
+  leading: false,
+  trailing: true,
+  context: undefined,
+};
+
+function getCoalesceConfig(
+  config: CoalesceConfig = defaultCoalesceConfig
+): Pick<CoalesceConfig, 'leading' | 'trailing'> & {
+  context: object | undefined;
+} {
+  return {
+    ...defaultCoalesceConfig,
+    ...config,
+  };
+}
+
+export const defaultCoalesceDurationSelector = <T>(value: T) =>
+  generateFrames();
+
+/**
+ * @description
```

Comment:
Please re-read your comments to correct some of the grammar.

Reply:
Done and rephreased.